src/share.rs

```diff
@@ -96,6 +96,19 @@ pub struct SendStream<B: IntoBuf> {
     inner: proto::StreamRef<B::Buf>,
 }
 
+/// A stream identifier, as described in [Section 5.1.1] of RFC 7540.
+///
+/// Streams are identified with an unsigned 31-bit integer. Streams
+/// initiated by a client MUST use odd-numbered stream identifiers; those
+/// initiated by the server MUST use even-numbered stream identifiers.  A
+/// stream identifier of zero (0x0) is used for connection control
+/// messages; the stream identifier of zero cannot be used to establish a
+/// new stream.
+///
+/// [Section 5.1.1]: https://tools.ietf.org/html/rfc7540#section-5.1.1
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
```

Comment:
I would recommend removing `Copy`. I'd also remove `Ord` and `PartialOrd` unless there is a case to keep them.

Reply:
The issue prompting this change (#289) states that:
> I'd expect `h2::StreamId` to implement at least copy and comparison traits.

I'm not sure whether or not @olix0r has any specific rationale or use-case behind implementing those traits, though?

Reply:
`Clone` would make more sense. `Copy` is pretty strong and adds a forwards compat hazard.